src/plans/D3MAaveDaiPlan.sol

```diff
@@ -0,0 +1,210 @@
+    function EXCESS_UTILIZATION_RATE() external view returns (uint256);
+    function variableRateSlope1() external view returns (uint256);
+    function variableRateSlope2() external view returns (uint256);
+    function baseVariableBorrowRate() external view returns (uint256);
+    function getMaxVariableBorrowRate() external view returns (uint256);
+}
+
+contract D3MAaveDaiPlan is ID3MPlan {
+
+    mapping (address => uint256) public wards;
+    InterestRateStrategyLike     public tack;
+    uint256                      public bar;  // Target Interest Rate [ray]
+
+    LendingPoolLike public immutable pool;
+    TokenLike       public immutable stableDebt;
+    TokenLike       public immutable variableDebt;
+    TokenLike       public immutable dai;
+    address         public immutable adai;
+
+    // --- Events ---
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event File(bytes32 indexed what, uint256 data);
+    event File(bytes32 indexed what, address data);
+
+    constructor(address dai_, address pool_) {
+        dai = TokenLike(dai_);
+        pool = LendingPoolLike(pool_);
+
+        // Fetch the reserve data from Aave
+        (,,,,,,, address adai_, address stableDebt_, address variableDebt_, address interestStrategy_,) = LendingPoolLike(pool_).getReserveData(dai_);
+        require(adai_ != address(0), "D3MAaveDaiPlan/invalid-adai");
+        require(stableDebt_ != address(0), "D3MAaveDaiPlan/invalid-stableDebt");
+        require(variableDebt_ != address(0), "D3MAaveDaiPlan/invalid-variableDebt");
+        require(interestStrategy_ != address(0), "D3MAaveDaiPlan/invalid-interestStrategy");
+
+        adai = adai_;
+        stableDebt = TokenLike(stableDebt_);
+        variableDebt = TokenLike(variableDebt_);
+        tack = InterestRateStrategyLike(interestStrategy_);
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    modifier auth {
+        require(wards[msg.sender] == 1, "D3MAaveDaiPlan/not-authorized");
+        _;
+    }
+
+    // --- Math ---
+    uint256 constant RAY  = 10 ** 27;
+    function _rmul(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = (x * y) / RAY;
+    }
+    function _rdiv(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = (x * RAY) / y;
+    }
+
+    // --- Admin ---
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    function file(bytes32 what, uint256 data) external auth {
+        if (what == "bar") {
+            bar = data;
+        } else revert("D3MAaveDaiPlan/file-unrecognized-param");
+        emit File(what, data);
+    }
+
+    function file(bytes32 what, address data) external auth {
+        if (what == "interestStrategy") tack = InterestRateStrategyLike(data);
```

Comment:
I think the convention here would be to have `what == "tack"`.

Reply:
aa7bada